core/src/HIP/Kokkos_HIP_Parallel_Range.hpp

```diff
@@ -555,10 +555,8 @@ class ParallelScanHIPBase {
       // correctly, the unit tests fail with wrong results
       const int gridMaxComputeCapability_2x = 0x01fff;
 
-      // FIXME_HIP block sizes greater than 256 don't work correctly,
-      // the unit tests fail with wrong results
       const int block_size =
-          std::min(static_cast<int>(local_block_size(m_functor)), 256);
+          std::min(static_cast<int>(local_block_size(m_functor)), 1024);
```

Comment:
Just remove the `std::min`. That was just a temporary hack.

Reply:
7943a27511ba51cc14cd7a2c172613abb6041c43